### PR TITLE
Implemented: empty state in search page(#201)

### LINKED ISSUE #201: Implement dedicated empty state image on the Search page
## What is the motivation for adding/enhancing this feature?
We need to implement an empty state which suggests user to take action for searching the product. 

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by tanishqueSharma**
### Related Issues

Closes #201

### Short Description and Why It's Useful

Implemented: Empty state in search page.

### Screenshots of Visual Changes before/after (If There Are Any)

Before
Screenshot (28)

After
Screenshot (29)

Screenshot (30)

### Contribution and Currently Important Rules Acceptance

- [x] I have read and followed [contribution rules](https://github.com/hotwax/inventory-count#contribution-guideline)

### CODE REVIEW

File: src/views/Search.vue [hunk 1]
```diff
@@ -7,6 +7,13 @@
     </ion-header>
     <ion-content>
       <ion-searchbar @ionFocus="selectSearchBarText($event)" v-model="queryString" :placeholder="$t('Search')" @keyup.enter="queryString = $event.target.value; searchProducts()"/>
+      <!-- Empty state -->
+      <div class="empty-state"  v-if="!products.length">
+        <!-- No result -->
+        <p v-if="products.length===0">{{ $t("No results found")}}</p>
```
Comment (sanskar345) [inline src/views/Search.vue:13]: v-if="products.length===0" is not required here, it is already in div with the condition. And improve the logic of displaying the "No results found" message. It should be diplayed only when the user searches for a product not initially.

File: src/store/modules/product/actions.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ const actions: ActionTree<ProductState, RootState> = {
         commit(types.PRODUCT_SEARCH_UPDATED, { products: products, totalProductsCount: totalProductsCount })
       } else if (payload.viewIndex == 0) {
         dispatch('clearSearchProducts')
-        showToast(translate("Product not found"));
+
```
Comment (sanskar345) [inline src/store/modules/product/actions.ts:37]: Remove the extra line

File: src/views/Search.vue [hunk 3]
```diff
@@ -143,7 +154,10 @@ export default defineComponent({
       }
       if (this.queryString) {
         await this.store.dispatch("product/findProduct", payload);
+        this.fetchingProduct = false;
```
Comment (sanskar345) [inline src/views/Search.vue:157]: No need assignment to fetchingProduct here it should be in searchProducts func

File: src/views/Search.vue [hunk 4]
```diff
@@ -143,7 +154,10 @@ export default defineComponent({
...
       } else {
+        this.fetchingProduct = false;
+        this.showErrorMessage = true;
```
Comment (sanskar345) [inline src/views/Search.vue:160]: No need of this assignment to var

**Review by sanskar345 — APPROVED** [review]

**Review by k2maan — APPROVED** [review]

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

**Pull request merged**